build.gradle

```diff
@@ -0,0 +1,87 @@
+subprojects {
+    apply plugin: 'java'
+    apply plugin: 'scala'
+
+    apply from: "${rootProject.projectDir}/gradle-scalatest.plugin"
+
+    ext {
+        ////////////////////////////////
+        // Commonly used dependencies //
+        ////////////////////////////////
+
+        mkDepGen = { genOne -> { String... names -> names.collect(genOne) } }
+        finagle = mkDepGen { "com.twitter:finagle-${it}_${scalaInterfaceVersion}:6.26.0" }
+        util = mkDepGen { "com.twitter:util-${it}_${scalaInterfaceVersion}:6.25.0" }
+        scrooge = mkDepGen { "com.twitter:scrooge-${it}_${scalaInterfaceVersion}:3.19.0" }
+        algebird = mkDepGen { "com.twitter:algebird-${it}_${scalaInterfaceVersion}:0.10.2" }
+        zipkin = mkDepGen { project(":zipkin-${it}") }
+
+        hadoop = mkDepGen { "org.apache.hadoop:hadoop-${it}:2.4.0" }
+        hadoopTest = { String... names -> hadoop(*names) + hadoop(*names).collect{"${it}:tests"} }
+
+        twitterZookeeperVersions = [
+                'candidate': "0.0.75",
+                'group'    : "0.0.81",
+                'client'   : "0.0.70",
+                'server-set': "1.0.103"
+        ]
+        zk = mkDepGen { "com.twitter.common.zookeeper:${it}:${twitterZookeeperVersions[it]}" }
+
+        twitterServer = "com.twitter:twitter-server_${scalaInterfaceVersion}:1.11.0"
+        junit = 'junit:junit:4.12'
+        slf4jLog4j12 = "org.slf4j:slf4j-log4j12:1.6.4"  // Dont forget to use for configuration 'runtime' only
+        ostrich = "com.twitter:ostrich_${scalaInterfaceVersion}:9.9.0"
+        griddle = 'com.yodle.griddle:griddle:1.7'
+    }
+
+    //////////////////////////
+    // Common configuration //
+    //////////////////////////
+
+    // SBT migration note: start of zipkinSettings
+    group "io.zipkin"
+    version zipkinVersion
+
+    repositories {
+        jcenter()
+        maven { url 'https://maven.twttr.com/' }
+        maven { url 'http://repo.typesafe.com/typesafe/releases/' }
+        // For zipkin-cassandra deps and com.twitter:kafka
+        maven { url 'http://conjars.org/repo' }
```

Comment:
I found that if project A depends on project B, and B has a dependency from resolver R, then A must also have that resolver, otherwise compiling A will fail, even if it's marked as being dependent on B. I see two solutions:

 1. Current setup
 1. Put the appropriate line into the `build.gradle` of each project where it's needed, of course DRY-ing it by declaring the URL once in a common place.

Did I miss something in the "problem statement"? Do you see another approach? Would you prefer the second solution?

I'll definitely document whatever the end result is in a comment, this is cause for confusion.